cmd/k8s-operator/proxygroup_specs.go

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !plan9
+
+package main
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	tsapi "tailscale.com/k8s-operator/apis/v1alpha1"
+	"tailscale.com/types/ptr"
+)
+
+const labelSecretType = "tailscale.com/secret-type"
+
+// Returns the base StatefulSet definition for a ProxyGroup. A ProxyClass may be
+// applied over the top after.
+func pgStatefulSet(pg *tsapi.ProxyGroup, namespace, cfgHash string) *appsv1.StatefulSet {
+	return &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            pg.Name,
+			Namespace:       namespace,
+			Labels:          labels("proxygroup", pg.Name, nil),
+			OwnerReferences: pgOwnerReference(pg),
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: ptr.To(pgReplicas(pg)),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels("proxygroup", pg.Name, nil),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:                       pg.Name,
+					Namespace:                  namespace,
+					Labels:                     labels("proxygroup", pg.Name, nil),
+					DeletionGracePeriodSeconds: ptr.To[int64](10),
+					Annotations: map[string]string{
+						podAnnotationLastSetConfigFileHash: cfgHash,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: pg.Name,
+					InitContainers: []corev1.Container{
+						{
+							Name:  "sysctler",
+							Image: fmt.Sprintf("tailscale/tailscale:%s", selfVersionImageTag()),
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: ptr.To(true),
+							},
+							Command: []string{
+								"/bin/sh",
+								"-c",
+							},
+							Args: []string{
+								"sysctl -w net.ipv4.ip_forward=1 && if sysctl net.ipv6.conf.all.forwarding; then sysctl -w net.ipv6.conf.all.forwarding=1; fi",
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "tailscale",
+							Image: fmt.Sprintf("tailscale/tailscale:%s", selfVersionImageTag()),
```

Comment:
(Here and for the init container) - we should respect the proxy image passed to helm install/via env vars.
It won't be that valuable for the initial release, but it would be better if we do it already now rather than on a subsequent release when it would can be breaking

Reply:
Or do you think it can only be configured via env vars?

Reply:
Agreed, thanks for catching this, I forgot to consider the env var.

Reply:
Fixed in e1d2b459b19e741d052aca13d3119f5e00960491